File: src/main/java/xyz/oribuin/skyblock/island/member/Member.java
package xyz.oribuin.skyblock.island.member;

import org.bukkit.Bukkit;
import org.bukkit.OfflinePlayer;
import xyz.oribuin.skyblock.nms.BorderColor;

import java.util.UUID;

public class Member {

    private final UUID uuid;
    private String username;
    private int island;
    private Role role;
    private BorderColor border;

    /**
     * Create a new instanceof Member with a UUID
     *
     * @param uuid The UUID of the player
     */
    public Member(UUID uuid) {
        this.uuid = uuid;
        this.username = "Unknown";
        this.island = -1;
        this.role = Role.MEMBER;
        this.border = BorderColor.BLUE;
    }

    /**
     * Get the player as an offline player
     *
     * @return The player object
     */
    public OfflinePlayer getOffline() {
        return Bukkit.getOfflinePlayer(this.uuid);
    }

    public UUID getUUID() {
        return uuid;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public int getIsland() {
        return island;
    }

    public void setIsland(int island) {
        this.island = island;
    }

    public Role getRole() {
        return role;
    }

    public void setRole(Role role) {
        this.role = role;
    }

    public BorderColor getBorder() {
        return border;
    }

    public void setBorder(BorderColor border) {
        this.border = border;
    }

}

File: Plugin/src/main/kotlin/xyz/oribuin/skyblock/island/Member.kt
package xyz.oribuin.skyblock.island

import java.util.*
import org.bukkit.Bukkit
import org.bukkit.OfflinePlayer
import org.bukkit.entity.Player
import xyz.oribuin.skyblock.nms.BorderColor

data class Member(val uuid: UUID) {
    var username: String = "Unknown" // The username of the player
    var island: Int = -1 // The Island ID
    var role = Role.MEMBER // The Role of the island member
    var border: BorderColor = BorderColor.BLUE // The color of the player's unique border color.

    enum class Role(val priority: Int) { // Island Roles and their priority
        OWNER(0),
        ADMIN(1),
        MEMBER(2),
        NONE(3)
    }

    // Does the user have an island.
    val hasIsland: Boolean
        get() = island >= 1

    // Get the member as a player.
    val offlinePlayer: OfflinePlayer
        get() = Bukkit.getOfflinePlayer(this.uuid)

    val onlinePlayer: Player?
        get() = this.offlinePlayer.player

}
